internal/storage/db_storage.go

```diff
@@ -0,0 +1,299 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgerrcode"
+	_ "github.com/jackc/pgx/v5/stdlib"
+	"log"
+	"net"
+	"time"
+)
+
+const (
+	Gauge   = "gauge"
+	Counter = "counter"
+)
+
+type DBStorage struct {
+	db *sql.DB
+}
+
+func (s *DBStorage) DB() *sql.DB {
+	return s.db
+}
+
+func NewDBStorage(dsn string) (*DBStorage, error) {
+	db, err := sql.Open("pgx", dsn)
```

Comment:
[CHANGE REQUEST] It should not be a part of this package. You can use any connection.